plan/preprocess.go

```diff
@@ -283,6 +294,33 @@ func (p *preprocessor) checkDropTableGrammar(stmt *ast.DropTableStmt) {
 	}
 }
 
+func (p *preprocessor) checkNonUniqTableAlias(stmt *ast.Join) {
+	if !p.isParentJoin {
+		p.tableAliasInJoin = append(p.tableAliasInJoin, make(map[string]interface{}))
+	}
+	tableAliases := p.tableAliasInJoin[len(p.tableAliasInJoin)-1]
+	if err := isTableAliasDuplicate(stmt.Left, tableAliases); err != nil {
+		p.err = err
+		return
+	}
+	if err := isTableAliasDuplicate(stmt.Right, tableAliases); err != nil {
+		p.err = err
+		return
+	}
+	p.isParentJoin = true
+}
+
+func isTableAliasDuplicate(node ast.ResultSetNode, tableAliases map[string]interface{}) error {
```

Comment:
How about s/isTableAliasDuplicate/checkDuplicatedTableAlias/, and making it a member function of `p`?

Reply:
`isTableAliasDuplicate ` is an utility function,  I name it follow the other utility functions in `preprocess.go` such as `isConstraintKeyTp`, `isInvalidDefaultValue`, `isIncorrectName`. I think it's better to keep this name and make it out of `p`

Reply:
ok